lib/algolia/client.rb

```diff
@@ -413,16 +429,20 @@ def thread_local_hosts(read)
       end
     end
 
-    def perform_request(session, url, method, data)
+    def perform_request(session, url, method, data, request_options)
+      hs = {}
+      extra_headers = request_options[:headers] || request_options['headers'] || {}
+      @headers.each { |key, val| hs[key.to_s] = val }
+      extra_headers.each { |key, val| hs[key.to_s] = val }
```

Comment:
Why do you explicitly call `key.to_s`? Coud we write `hs = @headers.merge(request_options[:headers] || request_options['headers'] || {})` instead?